### PR TITLE
Add microservice mailing

### PR DESCRIPTION
**Pull request by tsbonev**

### CODE REVIEW

File: bankapp/src/main/kotlin/com/clouway/bankapp/adapter/spark/RegisterController.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        support.addPropertyChangeListener(plc)
+    }
+
+    override fun handle(request: Request, response: Response): Any {
+        return try{
+            val user = userRepo
+                    .registerIfNotExists(
+                            transformer.fromJson(request.body(),
+                                    UserRegistrationRequest::class.java))
+            support.firePropertyChange("registration", null, user.username)
```
Comment (mlesikov) [inline bankapp/src/main/kotlin/com/clouway/bankapp/adapter/spark/RegisterController.kt]: userChangeListener.onRegistration()

listeners.forEach{it.onRegisration(user)}

listeners.forEach{it.onLogout(userId,userEmail)}



UserAsyncEventPublisher : UserChangeListener
UserChangeLogger:

client
asyncEventBus.publish(UserRegisteredEvent(userId, userName , email... )) 


microservice : 
/_ah/pubsubs/message -  PubSubsMessageHandler

messageEventBus.register(UserRegisrationHandler(), UserRegisteredEvent::class.java)

messageEventBus.register(UserLogaoutEventHandler(), UserLogaoutEvent::class.java)

Comment (tsbonev) [reply bankapp/src/main/kotlin/com/clouway/bankapp/adapter/spark/RegisterController.kt]: Changes applied.

**Comment by tsbonev** [comment]
PR is up to date.

File: mailservice/src/main/kotlin/com/clouway/mailservice/adapter/spark/UserRegistrationHandler.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import spark.Response
+
+/**
+ * @author Tsvetozar Bonev (tsbonev@gmail.com)
+ */
+class UserRegistrationHandler(private val mailer: Mailer) : EventHandler{
+    override fun handle(req: Request, res: Response): Any? {
+        return try {
+            val event = req.attribute<Event>("event")
+            mailer.mail((event as UserRegisteredEvent).username,
```
[the remark this answers is not in the document]
Comment (tsbonev) [reply mailservice/src/main/kotlin/com/clouway/mailservice/adapter/spark/UserRegistrationHandler.kt]: Could you elaborate on this? Are you referring to the class' name or the class' package?

[the remark this answers is not in the document]
Comment (qnislavnachev) [reply mailservice/src/main/kotlin/com/clouway/mailservice/adapter/spark/UserRegistrationHandler.kt]: My fault sorry. I was wrong about this.

**Pull request merged**